ONSdigital/ssdc-shared-events
# 3318 improve json schema example generation

# Motivation and Context

The `regen.sh` script was previously only capable of regenerating all examples at once, meaning every PR was bloated with unnecessary random data changing in every single example file. This change aims to provide a way to only target those examples which need regenerating. The caveat to this is that the entire `dictionary.md` must still be regenerated as one and this will still contain irrelevant bloat as it does not appear to be deterministically generated, orders can change when it is run twice with the same inputs.

# What has changed
- Support passing in schema file names as args to `regen.sh` to regenerate only specific examples.
- Update python polish script to support this change
- Update readme

# How to test?
Checkout this PR and try running `regen.sh` with no args. It should still function as before, regenerating all polished example json files. Reset this change, then try running it targetting specific schema files. It should then only regenerate those targetted files, plus the dictionary.

# Links
https://trello.com/c/12HfbSbL/3282-rm-documentation-infrastructure-overview-5
